### PR TITLE
Implement .NET 7 runtime support and bindings generation

### PR DESCRIPTION
**Pull request by GoaLitiuM**
Trello checklist in roadmap: https://trello.com/c/xgS3TphU/351-net-7

This is a mostly functional prototype of the engine running in .NET 7 runtime (Windows-only for now). The editor should be able to run the Flax Samples without any issues, and cooked builds should also be working. Not all features have been tested fully, so some errors can be expected.

The engine is still expected to be built with Mono header files included, as some of the Mono-related functions are called directly throughout the engine. The prototype wraps the Mono-functions and calls functionally equivalent methods in managed side using `nethost` library for hosting the .NET runtime in the same process. In the future, the engine should avoid calling Mono-functions directly and instead use the already established wrapper classes in `Engine/Scripting/ManagedCLR` to communicate with managed runtime.

### CODE REVIEW

**Comment by GoaLitiuM** [comment]
Disabled the other Github actions for now and only running the Windows-related workflows.

File: Source/Tools/Flax.Build/Projects/VisualStudio/VCProjectGenerator.cs [hunk 1]
```diff
@@ -175,9 +176,9 @@ public override void GenerateProject(Project project)
                 vcProjectFileContent.AppendLine("    <LibraryWPath />");
                 vcProjectFileContent.AppendLine("    <SourcePath />");
                 vcProjectFileContent.AppendLine("    <ExcludePath />");
-                vcProjectFileContent.AppendLine(string.Format("    <NMakeBuildCommandLine>{0} -build</NMakeBuildCommandLine>", cmdLine));
-                vcProjectFileContent.AppendLine(string.Format("    <NMakeReBuildCommandLine>{0} -rebuild</NMakeReBuildCommandLine>", cmdLine));
-                vcProjectFileContent.AppendLine(string.Format("    <NMakeCleanCommandLine>{0} -clean</NMakeCleanCommandLine>", cmdLine));
+                vcProjectFileContent.AppendLine(string.Format("    <NMakeBuildCommandLine>dotnet {0} -build</NMakeBuildCommandLine>", cmdLine));
```
Comment (mafiesto4) [inline Source/Tools/Flax.Build/Projects/VisualStudio/VCProjectGenerator.cs:179]: Why not run the build tool directly here?

File: Source/Engine/Core/Types/Variant.cpp [hunk 2]
```diff
@@ -1578,8 +1590,10 @@ Variant::operator void*() const
     case VariantType::Blob:
         return AsBlob.Data;
     case VariantType::ManagedObject:
-#if USE_MONO
-        return AsUint ? mono_gchandle_get_target(AsUint) : nullptr;
+#if USE_NETCORE
+        return AsUint64 ? MUtils::GetGCHandleTarget(AsUint64) : nullptr;
```
Comment (mafiesto4) [inline Source/Engine/Core/Types/Variant.cpp:1594]: Maybe we could just use `AsUint64` for GCHandleTarget storage for both Mono and NetCore versions? (simplify code)

File: Source/Engine/Main/Windows/main.cpp [hunk 3]
```diff
@@ -50,14 +50,14 @@ int APIENTRY wWinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance, LPTSTR lpCmd
 #endif
 
     Platform::PreInit(hInstance);
-    __try
+    //__try
```
Comment (mafiesto4) [inline Source/Engine/Main/Windows/main.cpp:53]: NetCore catches C++ exceptions?

File: Source/Engine/Scripting/Types.h [hunk 4]
```diff
@@ -49,6 +49,17 @@ typedef void MObject;
...
 
+#if USE_NETCORE
+struct _MonoDomain {};
+struct _MonoThread {};
+#endif
+
+#if USE_NETCORE
+typedef unsigned long long gchandle;
+#else
+typedef uint32 gchandle;
```
Comment (mafiesto4) [inline Source/Engine/Scripting/Types.h:60]: Maybe we will simplfy gchandle to uint64 so Mono runtime will do casting but we will get simpler code in engine?

**Review by mafiesto4 — COMMENTED** [review]
Great stuff!

File: Source/Tools/Flax.Build/Projects/VisualStudio/VCProjectGenerator.cs (hunk 1 above)
Comment (GoaLitiuM) [reply Source/Tools/Flax.Build/Projects/VisualStudio/VCProjectGenerator.cs:179]: I think this was leftover code from the early versions where the build tool got built as a DLL-only, and you'd need to use `dotnet` to call DLL-binaries.

File: Source/Engine/Main/Windows/main.cpp (hunk 3 above)
Comment (GoaLitiuM) [reply Source/Engine/Main/Windows/main.cpp:53]: More leftover code from debugging

**Comment by mafiesto4** [comment]
Merged!  Works great! Thanks :) Now, time to work on quality and matching all the features. I'll take platforms support on myself.

Progress: https://trello.com/b/NQjLXRCP/flax-roadmap

**Pull request merged**